### PR TITLE
[ws-manager] Wait additional time to deletionGracePeriod before finalize workspace

### PR DESCRIPTION
**Pull request by aledbf**
## Description

> This error happens quite often (> 6000 entries in the last 7 days) and seems to be user-facing (i.e. error message on start workspace).

From the detail of the several traces, the error happens in ring 0 during the shutdown (there is only two connection to InWorkspaceService). We are too eager to call `finalizeWorkspaceContent`.
This PR adds an additional wait time on top of the `deletionGracePeriod` to make sure the changes propagate on the data plane.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/5574

## How to test
- start a workspace
- kill `ws-daemon` pod
- wait until new `ws-daemon` pod is running
- stop workspace
- check there are no errors in `ws-daemon` removing workspace content and
- check the backup was created

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **aledbf** after the PR has been reviewed.

Associated issue: *#5574*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ws-manager/pkg/manager/monitor.go [hunk 1]
```diff
@@ -359,14 +359,26 @@ func actOnPodEvent(ctx context.Context, m actingManager, status *api.WorkspaceSt
...
+			var gracePeriod int64 = 30
+			if wso.Pod.DeletionGracePeriodSeconds != nil {
+				gracePeriod = *wso.Pod.DeletionGracePeriodSeconds
+			}
+			ttl := time.Duration(gracePeriod) * time.Second * 2
+
+			go func() {
+				time.Sleep(ttl)
+				m.finalizeWorkspaceContent(ctx, wso)
+			}()
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/monitor.go:381]: Do we need to ensure we're not starting too many go routines here? We'll enter `stopping` several times.

Comment (aledbf) [reply components/ws-manager/pkg/manager/monitor.go:381]: Not sure that's required. We know for how long the goroutine will run, and calling `finalizeWorkspaceContent` multiple times has no side effects.

Also, adding control over the number of goroutines would imply keeping some kind of state.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: bb2c11f4ae1e3cc186e1c04fb697699b7c095112</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5610#issuecomment-916675083" title="LGTM">csweichel</a>*

Associated issue: *#5574*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**